.github/workflows/ci.yml

```
@@ -108,25 +108,34 @@ jobs:
           Cython==0.29.* --install-option="--no-cython-compile"
         echo "$(python3 -m site --user-base)/bin" >> $GITHUB_PATH
 
+
+    # GitHub actions currently does not support modifying an already existing
+    # cache. Hence, we create a new cache for each commit with key
+    # cache-${{ runner.os }}-${{ matrix.cache-key }}-${{ github.sha }}. This
+    # will result in an initial cache miss. However, restore-keys will search
+    # for the most recent cache with prefix
+    # cache-${{ runner.os }}-${{ matrix.cache-key }}-, and if found uses it as
+    # a base for the new cache.
+
     - name: Restore Dependencies
       id: restore-deps
       uses: actions/cache@v1
+      with:
+        path: build/deps
+        key: ${{ runner.os }}-deps-${{ hashFiles('cmake/Find**') }}-${{ hashFiles('.github/workflows/ci.yml') }}
+
+    - name: Restore Auxiliary Tools
+      id: restore-aux-tools
+      uses: actions/cache@v1
```

```suggestion
      uses: actions/cache@v2
```

Please also update the other cache versions to v2.